File: server/API/Models/DTOs/PostDto.cs
namespace API.Models.DTOs
{
    public class PostDto
    {
        public int PostId { get; set; }
        public string Content { get; set; }
        
        public int LearningResourceId { get; set; }
        public LearningResourceDto LearningResource { get; set; }
    }
}

File: client/src/app/shared/models/dtos/post_dto.ts
import { UserPostAction } from 'src/app/+learning/models';
import { LearningResource } from './learning_resource_dto';

export interface Post {
  postId?: string;
  content?: string;
  learningResourceId?: string;
  userId?: string;
  learningResource?: LearningResource;
  userPostAction?: UserPostAction;
  previousUserPostAction?: UserPostAction;
  likes?: number;
  reports?: number;
}
